File: client/src/types/order.ts
export type OrderStatus = 'waiting' | 'accepted' | 'in_progress' | 'completed' | 'cancelled';

/** Форма відповіді бекенду (`OrderResponse` у server/TaxiSystem.Api/DTOs/OrderDtos.cs). */
export interface ApiOrder {
  order_id: string;
  passenger_email: string;
  passenger_name: string;
  pickup_location: string;
  destination: string;
  car_class: string;
  estimated_cost: number;
  weather_hazard_level: 'HIGH' | 'NORMAL';
  current_status: OrderStatus;
  safe_route_applied: boolean;
  payment_method: string;
  payment_status: string;
  driver_email: string | null;
  driver_name: string | null;
  end_time: string | null;
}

File: server/TaxiSystem.Api/Models/Order.cs
namespace TaxiSystem.Api.Models;


public class Order
{
    public string OrderId { get; set; } = string.Empty;

    public string PassengerEmail { get; set; } = string.Empty;
    public string PassengerName { get; set; } = string.Empty;

    public string PickupLocation { get; set; } = string.Empty;
    public string Destination { get; set; } = string.Empty;
    public string CarClass { get; set; } = string.Empty;

    public double? PickupLat { get; set; }
    public double? PickupLng { get; set; }
    public double? DestinationLat { get; set; }
    public double? DestinationLng { get; set; }

    public int EstimatedCost { get; set; }

    /// <summary>"HIGH" | "NORMAL"</summary>
    public string WeatherHazardLevel { get; set; } = "NORMAL";

    /// <summary>"waiting" | "accepted" | "in_progress" | "completed"</summary>
    public string CurrentStatus { get; set; } = "waiting";

    public bool SafeRouteApplied { get; set; }

    public string PaymentMethod { get; set; } = string.Empty;
    public string PaymentStatus { get; set; } = string.Empty;

    public string? DriverEmail { get; set; }
    public string? DriverName { get; set; }
    public string? EndTime { get; set; }

    /// <summary>1-5, ставить лише пасажир і лише після "completed". Null, поки не оцінено.</summary>
    public int? Rating { get; set; }

    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
}
